### PR TITLE
Change to using templates for docker and compose files.

### PR DESCRIPTION
**Pull request by NCarlsonMSFT**

### CODE REVIEW

File: generators/app/aspnetHelper.js [hunk 1]
```diff
@@ -7,48 +7,26 @@ var util = require('./utils.js');
...
-        _dockerfileHelper.addEntrypointCommand('["dotnet", "run"]');
-    } else {
-        _dockerfileHelper.addEntrypointCommand('["dnx", "-p", "project.json", "web"]');
+    this._portNumber = portNumber
+    
+    switch (baseImageName) {
+        case 'aspnet:1.0.0-rc1-update1':
+            this._templateFolder = 'dnx';
+            break;
+        case 'dotnet:1.0.0-preview1':
```
Comment (PeterJausovec) [inline generators/app/aspnetHelper.js]: I'd explicitly set the value for this._templateFolder

File: generators/app/templates/docker-compose.yml [hunk 2]
```diff
@@ -0,0 +1,9 @@
+<%= imageName %>:
+  dockerfile: Dockerfile.<%= environment %>
+  build: .<% if (isWebProject && portNumber) { %>
```
Comment (PeterJausovec) [inline generators/app/templates/docker-compose.yml]: new line to make it easier to read?

**Comment by PeterJausovec** [comment]
:shipit:

File: generators/app/templates/docker-compose.yml (hunk 2 above)
Comment (NCarlsonMSFT) [inline generators/app/templates/docker-compose.yml]: If i put the newline outside of the tag it ends up in the generated file. Would you like one within the tag?

---
In reply to: [64833362](https://github.com/Microsoft/generator-docker/pull/57#discussion_r64833362) [](ancestors = 64833362)

File: generators/app/templates/dotnet/Dockerfile [hunk 3]
```diff
@@ -0,0 +1,8 @@
+FROM <%= baseImageName %>
+COPY project.json /app/
+WORKDIR /app
+RUN ["dotnet", "restore"]
+COPY . /app
+RUN ["dotnet", "build", "-c", "<%= environment %>"]
+EXPOSE <%= portNumber %>
+ENTRYPOINT ["dotnet", "run"]
```
Comment (saurabsa) [inline generators/app/templates/dotnet/Dockerfile]: Are we just targeting deploying? No Edit-Refresh or Debugging?

**Comment by saurabsa** [comment]
>  * Checks if  'web' command is in the  project.json and adds it if command is not there yet.

space

---
Refers to: generators/app/aspnetHelper.js:64 in d668eec. [](commit_id = d668eecf58b8578ec4941b133bfd823384ed40d5, deletion_comment = False)

File: generators/app/templates/dotnet/Dockerfile (hunk 3 above)
Comment (NCarlsonMSFT) [inline generators/app/templates/dotnet/Dockerfile]: Yes, these are just fro getting an image built and a container running.

---
In reply to: [64834074](https://github.com/Microsoft/generator-docker/pull/57#discussion_r64834074) [](ancestors = 64834074)

**Comment by saurabsa** [comment]
>                 data = data.replace('new WebHostBuilder()', 'new WebHostBuilder().UseUrls("http://*:' + port + '")');

With this change, don't we need to set ENV ASPNETCORE_URLS [http://*:80](http://*:80) in the dotnet Dockerfile?

---
Refers to: generators/app/aspnetHelper.js:82 in d668eec. [](commit_id = d668eecf58b8578ec4941b133bfd823384ed40d5, deletion_comment = False)

File: generators/app/templates/dotnet/Dockerfile [hunk 4]
```diff
@@ -0,0 +1,8 @@
+FROM <%= baseImageName %>
+COPY project.json /app/
+WORKDIR /app
+RUN ["dotnet", "restore"]
```
Comment (saurabsa) [inline generators/app/templates/dotnet/Dockerfile]: aspnetHelper.js included updating Program.cs to inculde UseUrls. Shouldn't ENV ASPNETCORE_URLS [http://*:80](http://*:80) be added here for that change?

File: generators/app/templates/dotnet/Dockerfile [hunk 5]
```diff
@@ -0,0 +1,8 @@
+FROM <%= baseImageName %>
```
Comment (saurabsa) [inline generators/app/templates/dotnet/Dockerfile]: aspnetHelper.js included updating Program.cs to inculde UseUrls. Shouldn't ENV ASPNETCORE_URLS [[http://*:80](http://*:80)]([http://*:80](http://*:80)) be added here for that change?

Comment (saurabsa) [inline generators/app/templates/dotnet/Dockerfile]: aspnetHelper.js included updating Program.cs to inculde UseUrls. Shouldn't ENV ASPNETCORE_URLS [[http://*:80](http://*:80)]([http://*:80](http://*:80)) be added here for that change?

File: generators/app/templates/dotnet/Dockerfile [hunk 6]
```diff
@@ -0,0 +1,8 @@
...
+COPY project.json /app/
+WORKDIR /app
+RUN ["dotnet", "restore"]
+COPY . /app
+RUN ["dotnet", "build", "-c", "<%= environment %>"]
```
Comment (saurabsa) [inline generators/app/templates/dotnet/Dockerfile]: aspnetHelper.js included updating Program.cs to inculde UseUrls. Shouldn't ENV ASPNETCORE_URLS [[http://*:80](http://*:80)]([http://*:80](http://*:80)) be added here for that change?

File: generators/app/templates/dotnet/Dockerfile (hunk 5 above)
Comment (saurabsa) [inline generators/app/templates/dotnet/Dockerfile]: aspnetHelper.js included updating Program.cs to inculde UseUrls. Shouldn't ENV ASPNETCORE_URLS [http://*:80](http://*:80) be added here for that change?

**Comment by saurabsa** [comment]
:shipit:

**Pull request merged**